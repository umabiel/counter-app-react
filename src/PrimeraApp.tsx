import { Fragment, ReactNode } from "react";
import PropTypes from "prop-types";

interface IProps {
  titulo?: ReactNode;
  subtitulo?: ReactNode;
}

// FC
const PrimeraApp = ({ titulo, subtitulo }: IProps) => {
  return (
    <Fragment>
      <h1> {titulo} </h1>
      <p>SubTitulo: {subtitulo}</p>
    </Fragment>
  );
};

PrimeraApp.propTypes = {
  titulo: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  titulo: "NONE",
};

export default PrimeraApp;
