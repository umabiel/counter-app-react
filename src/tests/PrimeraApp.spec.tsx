import Enzyme, { shallow } from "enzyme";
import ReactSeventeenAdapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";

import PrimeraApp from "../PrimeraApp";

Enzyme.configure({ adapter: new ReactSeventeenAdapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Pruebas en componente <PrimerApp />", () => {
  //   test('Debe de mostrar el mensaje "Hola Gustavo"', () => {
  //     const saludo = "Hola Gustavo";
  //     const wrapper = render(<PrimeraApp titulo={saludo} />);
  //     expect(wrapper.getByText(saludo)).toBeInTheDocument();
  //   });

  test("Debe de mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "Hola Gustavo";
    const wrapper = shallow(<PrimeraApp titulo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe renderizar el subtitulo", () => {
    const saludo = "Hola Gustavo";
    const subtitutlo = "SubaruForester";
    const wrapper = shallow(
      <PrimeraApp titulo={saludo} subtitulo={subtitutlo} />
    );
    const textoParrafo = wrapper.find("p").text();
    expect(textoParrafo).toBe(subtitutlo);
  });
});
