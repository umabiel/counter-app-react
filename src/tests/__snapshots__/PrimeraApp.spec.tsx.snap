// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <PrimerApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola Gustavo
     
  </h1>
  <p />
</Fragment>
`;
